assets/js/blocks/stock-filter/utils.ts

```diff
@@ -28,6 +28,18 @@ export const getActiveFilters = (
 	);
 };
 
+export function generateUniqueId() {
+	return Math.floor( Math.random() * Date.now() );
+}
+
+export const formatSlug = ( slug: string ) =>
+	slug
+		.trim()
+		.replace( /\s/g, '' )
+		.replace( /_/g, '-' )
+		.replace( /-+/g, '-' )
+		.replace( /[^a-zA-Z0-9-]/g, '' );
```

Comment:
Right, `.replace( /\s/g, '' ) can be merged with `.replace( /[^a-zA-Z0-9-]/g, '' );`.

About the `.replace( /_/g, '-' ).replace( /-+/g, '-' )` is not equal to `.replace( /_|-+/g, '-' )` unless you had something else in mind? In case of two following operations the sequence of dashes and underscores will be shrinked to single character, in the latter it won't. 🤔 But regex is not my strongest field, I may totally be missing something 😅